db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb

```diff
@@ -1,70 +1,119 @@
 #
-# Description: This method checks to see if the stack has been provisioned
+# Description: This class checks to see if the stack has been provisioned
 #   and whether the refresh has completed
 #
+module ManageIQ
+  module Automate
+    module Cloud
+      module Orchestration
+        module Provisioning
+          module StateMachines
+            class CheckProvisioned
+              def initialize(handle = $evm)
+                @handle = handle
+              end
 
-def refresh_provider(service)
-  provider = service.orchestration_manager
+              def main
+                task = @handle.root["service_template_provision_task"]
+                service = task.try(:destination)
 
-  $evm.log("info", "Refreshing provider #{provider.name}")
-  $evm.set_state_var('provider_last_refresh', provider.last_refresh_date.to_i)
-  provider.refresh
-end
+                unless service && service.instance_of?(MiqAeMethodService::MiqAeServiceServiceOrchestration)
+                  @handle.log(:error, 'Service does not exist or has a different' \
+                                      ' type from MiqAeServiceServiceOrchestration')
+                  exit(MIQ_STOP)
```

Comment:
@pkomanek 
Should we check for the service first and then check the type of service.
Otherwise for nil we would be saying it is a different kind of service